modules/asset-cdn/asset-cdn.php

```diff
@@ -0,0 +1,227 @@
+			$urls[] = str_replace( untrailingslashit( $site_url ), '', $script->src );
+			$vers[] = $script->ver ? $script->ver : $wp_scripts->default_version;
+		}
+
+		$cdn_url = $this->cdn_server . '/js?b=' .
+			urlencode( $site_url ) . '&' .
+			http_build_query( array( 'f' => $urls ) ) . '&' .
+			http_build_query( array( 'v' => $vers ) );
+		// TODO: if there is NO inline or external script tags in the body, render async (maybe?)
+		echo '<script type="text/javascript" src="' . esc_attr( $cdn_url ) . '"></script>';
+	}
+
+	/**
+	 * Asset modification functions
+	 */
+
+	/**
+	 * Scripts
+	 */
+
+	public function register_concat_scripts( $tag, $handle, $src ) {
+		global $wp_scripts;
+
+		// don't do admin for now
+		if ( is_admin() || ! isset( $wp_scripts->registered[$handle] ) ) {
+			return $tag;
+		}
+
+		$script = $wp_scripts->registered[$handle];
+
+		if ( $this->should_concat_script( $script ) ) {
+			$this->buffer_script( $script );
+			return '';
+		}
+
+		return $tag;
+	}
+
+	private function should_concat_script( $script ) {
+		// only concat local scripts
+		$is_local       = $this->is_local_url( $script->src );
+		// don't concat conditional scripts
+		$is_conditional = isset( $script->extra['conditional'] );
+		return apply_filters( 'jetpack_perf_concat_script', $is_local && ! $is_conditional, $script->handle, $script->src );
+	}
+
+	private function buffer_script( $script ) {
+		$group = isset( $script->extra['group'] ) ? $script->extra['group'] : 0;
+		if ( ! isset( $this->concat_script_groups[$group] ) ) {
+			$this->concat_script_groups[$group] = array();
+		}
+		$this->concat_script_groups[$group][] = $script;
+	}
+
+	/**
+	 * Styles
+	 */
+
+	public function register_concat_styles( $tag, $handle, $href, $media ) {
+		global $wp_styles;
+
+		// don't do admin for now
+		if ( is_admin() || ! isset( $wp_styles->registered[$handle] ) ) {
+			return $tag;
+		}
+
+		$style = $wp_styles->registered[$handle];
+
+		if ( $this->should_concat_style( $style ) ) {
+			$this->buffer_style( $style );
+			return '';
+		}
+
+		return $tag;
+	}
+
+	private function buffer_style( $style ) {
+		$group = isset( $style->extra['group'] ) ? $style->extra['group'] : 0;
+		$media = $style->args;
+		if ( ! $media ) {
+			$media = 'all';
+		}
+		if ( ! isset( $this->concat_style_groups[$group] ) ) {
+			$this->concat_style_groups[$group] = array();
+		}
+		if ( ! isset( $this->concat_style_groups[$group][$media] ) ) {
+			$this->concat_style_groups[$group][$media] = array();
+		}
+		$this->concat_style_groups[$group][$media][] = $style;
+	}
+
+	private function should_concat_style( $style ) {
+		// only concat local styles
+		$is_local       = $this->is_local_url( $style->src );
+		// don't concat conditional styles
+		$is_conditional = isset( $style->extra['conditional'] );
+		return apply_filters( 'jetpack_perf_concat_style', $is_local && ! $is_conditional, $style->handle, $style->src );
+	}
+
+	private function is_local_url( $url ) {
```

Comment:
Do we maybe have something in Photon that can help with this?

Reply:
Couldn't find anything, alas